src/app/(sidebar)/transactions-explorer/page.tsx

```diff
@@ -45,9 +70,18 @@ export default function Explorer() {
         await txsQuery.refetch();
 
         if (txsQuery.data?.transactions && !txsQuery.error) {
-          txsQuery.data.transactions.forEach((tx) => {
-            transactions.set(tx.txHash, tx);
-          });
+          for (const txinfo of txsQuery.data.transactions) {
+            const normalizedTx = await normalizeTransaction(txinfo);
+
+            if (normalizedTx) {
+              transactions.set(normalizedTx.txHash, normalizedTx);
+            }
+          }
+
+          const txs = Array.from(transactions.values())
+            .toSorted((tx) => tx.createdAt)
+            .splice(-500);
```

Comment:
just curious - what does this `-500` do

Reply:
negative indexes work like `len - 500` with splice 🤓

Reply:
why `.splice` over `.slice`?

Reply:
in this particular case, both would work interchangeably.